### PR TITLE
Extracted certificate installation

### PR DESCRIPTION
**Pull request by lpusok**
### Context

Certificate installation is a common part of both Bitrise-managed and Apple-managed code signing, extracted it to a common phase.
Resolves: https://bitrise.atlassian.net/browse/STEP-1932

### CODE REVIEW

File: autocodesign/autocodesign.go [hunk 1]
```diff
@@ -107,12 +107,17 @@ type CertificateProvider interface {
...
+type LocalCertificates map[appstoreconnect.CertificateType][]certificateutil.CertificateInfoModel
+
+// CodesignAssetsOpts are codesigning related parameters that are not specified by the project (or archive)
 type CodesignAssetsOpts struct {
-	DistributionType       DistributionType
-	BitriseTestDevices     []devportalservice.TestDevice
-	MinProfileValidityDays int
-	VerboseLog             bool
+	DistributionType                  DistributionType
+	TypeToBitriseCertificates         LocalCertificates
```
Comment (godrei) [inline autocodesign/autocodesign.go:116]: Why is it `BitriseCertificates` ?

File: codesign/codesign.go [hunk 2]
```diff
@@ -250,22 +260,45 @@ func (m *Manager) selectCodeSigningStrategy(credentials appleauth.Credentials) (
...
+		// Empty passphrase provided, as already parsed certificate + private key
+		if err := m.assetInstaller.InstallCertificate(cert); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (m *Manager) checkXcodeManagedCertificates(certificates []certificateutil.CertificateInfoModel) error {
```
Comment (godrei) [inline codesign/codesign.go:291]: I do not see how the function name relates to the function body. I mean I do not get why the XcodeManaged part.

File: codesign/codesign_test.go [hunk 3]
```diff
@@ -2,10 +2,11 @@ package codesign
 
 import (
 	"errors"
-	"github.com/bitrise-io/go-xcode/v2/codesign/mocks"
 	"testing"
 	"time"
 
+	"github.com/bitrise-io/go-xcode/v2/codesign/mocks"
+
```
Comment (godrei) [inline codesign/codesign_test.go:9]: extra newline

File: autocodesign/autocodesign.go (hunk 1 above)
Comment (lpusok) [reply autocodesign/autocodesign.go:116]: LocalCertificates

File: codesign/codesign.go [hunk 4]
```diff
@@ -319,18 +343,42 @@ func (m *Manager) prepareCodeSigningWithBitrise(credentials appleauth.Credential
...
+	}
+
+	if len(certs) > 0 {
+		m.logger.Printf("%d certificates downloaded:", len(certs))
+		for _, cert := range certs {
+			m.logger.Printf("- %s", cert.String())
+		}
+	} else {
+		m.logger.Warnf("No certificates are uploaded to Bitrise.")
+	}
```
Comment (lpusok) [inline codesign/codesign.go:361]: use downloadCertifiactes

**Review by godrei — APPROVED** [review]

**Pull request merged**